h/api/uri.py

```diff
@@ -140,11 +141,11 @@ def normalize(uristr):
 
     # If this isn't a URL, we don't perform any normalization
     if uri.scheme.lower() not in URL_SCHEMES:
-        return uristr
+        return text_type(uristr)
```

Comment:
This will fail on Python 2 if `uristr` has non-ASCII characters in it. You should specify the encoding just as you've been doing inline.